components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/AbstractAPIManager.java

```diff
@@ -686,6 +687,17 @@ protected final void handleResourceAlreadyExistsException(String msg) throws API
         log.error(msg);
         throw new APIMgtResourceAlreadyExistsException(msg);
     }
+
+    @Override
+    public WorkflowResponse completeWorkflow(WorkflowExecutor workflowExecutor, Workflow workflow)
+            throws APIManagementException {
+        if (workflow.getWorkflowReference() == null) {
+            String message = "Error while changing the workflow. Missing reference";
+            log.error(message);
+            throw new APIManagementException(message, ExceptionCodes.WORKFLOW_EXCEPTION);
+        }
+        return workflow.completeWorkflow(workflowExecutor);
+    }
```

Comment:
need to add a restriction here to only allow publisher or store side workflow completion based on which rest api it get called (if complete is called from publisher's callback rest api, only execute publisher's workflow)